src/Compilers/CSharp/Portable/Utilities/ValueSetFactory.PositiveIntTC.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    using static BinaryOperatorKind;
+
+    internal static partial class ValueSetFactory
+    {
+        private struct PositiveIntTC : INumericTC<int>
```

Comment:
I wonder if someday in the future we'll be able to get rid of `INumericTC` and just use the generic math built into .NET...

Reply:
We should already be able to use statics in interfaces at least, probably not until we want to drop netframework support?